dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java

```diff
@@ -245,4 +252,87 @@ private static HTMLPageAsset createPage(){
         ContentletDataGen.publish(htmlPageAsset);
         return  htmlPageAsset;
     }
+
+    /**
+     * Method to test: {@link VelocityServlet#service(HttpServletRequest, HttpServletResponse)}
+     *
+     * When: TimeMachine in running and request a existing Page
+     * Should: Return the page
+     */
+    @Test
+    public void testingTimeMachine () throws Exception{
+
+        final User systemUser = APILocator.systemUser();
+        final ContentType contentGenericType = APILocator.getContentTypeAPI(systemUser).find("webPageContent");
+
+        Container container = new ContainerDataGen().site(host).nextPersisted();
+        final Template template = new TemplateDataGen().site(host).withContainer(container.getIdentifier()).nextPersisted();
+
+        final List<ContainerStructure> csList = new ArrayList<ContainerStructure>();
+        final ContainerStructure cs = new ContainerStructure();
+        cs.setStructureId(contentGenericType.id());
+        cs.setCode("$!{body}");
+        csList.add(cs);
+
+        container = APILocator.getContainerAPI().save(container, csList, host, systemUser, false);
+        PublishFactory.publishAsset(container, systemUser, false, false);
+
+        Config.setProperty("DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE", false);
+        Config.setProperty("DEFAULT_PAGE_TO_DEFAULT_LANGUAGE", false);
+
+        final Contentlet page = new HTMLPageDataGen(host, template).host(host).languageId(1).nextPersisted();
+
+        final Contentlet contentlet = new ContentletDataGen(contentGenericType.id())
+                .languageId(1)
+                .host(host)
+                .setProperty("title", "content1")
+                .setProperty("body", "content1")
+                .nextPersisted();
+
+        ContentletDataGen.publish(contentlet);
+        ContentletDataGen.publish(page);
+
+        new MultiTreeDataGen()
+                .setContainer(container)
+                .setPage((HTMLPageAsset) page)
+                .setContentlet(contentlet)
+                .nextPersisted();
+
+        final HttpServletRequest mockRequest = new MockSessionRequest(
+                new MockAttributeRequest(new MockHttpRequest(host.getName(), ((HTMLPageAsset) page).getURI()).request()).request()
+        )
+                .request();
+
+
+        Mockito.when(mockRequest.getParameter("host_id")).thenReturn(host.getIdentifier());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Mockito.when' due to existing static import 'org.mockito.Mockito.when'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5474742)